In [1]:
import pandas as pd 

data = pd.read_csv('chatGPTPT1.csv')
data = data.drop('Unnamed: 0', axis=1)
data = data.drop('Entity', axis=1)
data

Label   
0                 David Bowie  \
1                  Kanye West   
2                      Eminem   
3            Jennifer Aniston   
4              George W. Bush   
..                        ...   
179              Nancy Pelosi   
180            Andhra Pradesh   
181               Maharashtra   
182  Paris Saint-Germain F.C.   
183             Uttar Pradesh   

                                              Abstract   
0    David Robert Jones (8 January 1947 – 10 Januar...  \
1    Ye (/jeɪ/ YAY; born Kanye Omari West /ˈkɑːnjeɪ...   
2    Marshall Bruce Mathers III (born October 17, 1...   
3    Jennifer Joanna Aniston (born February 11, 196...   
4    George Walker Bush (born July 6, 1946) is an A...   
..                                                 ...   
179  Nancy Patricia Pelosi (/pəˈloʊsi/; née D'Alesa...   
180  Andhra Pradesh (English: /ˌɑːndrə prəˈdɛʃ/, Te...   
181  Maharashtra (/məhɑːˈrɑːʃtrə/; Marathi: [məhara...   
182  Paris Saint-Germain Football Club (French pron...   
183  Uttar Pradesh (/ˌʊtər prəˈdɛʃ/; Hindi: [ˈʊtːəɾ...   

                     Subjects   
0                 David_Bowie  \
1                  Kanye_West   
2                      Eminem   
3            Jennifer_Aniston   
4              George_W._Bush   
..                        ...   
179              Nancy_Pelosi   
180            Andhra_Pradesh   
181               Maharashtra   
182  Paris_Saint-Germain_F.C.   
183             Uttar_Pradesh   

                                       GPT-Triples-Raw  
0    @prefix ex: <http://example.com/> .\n@prefix r...  
1    @prefix ex: <http://example.com/> .\n@prefix r...  
2    @prefix ex: <http://example.com/> .\n@prefix r...  
3    ```\n@prefix ex: <http://example.com/> .\n@pre...  
4    @prefix ex: <http://example.com/> .\n@prefix x...  
..                                                 ...  
179  @prefix ex: <http://example.com/> .\n\nex:Nanc...  
180  @prefix ex: <http://example.com/> .\n\nex:Andh...  
181  @prefix ex: <http://example.com/> .\n@prefix r...  
182  @prefix ex: <http://example.com/> .\n\nex:Pari...  
183  @prefix ex: <http://example.com/> .\n@prefix r...  

[184 rows x 4 columns]

In [2]:
import random 
random.seed(999)
nums = random.sample(range(0, 183), 10)
nums

[173, 20, 145, 146, 136, 125, 123, 33, 165, 81]

In [102]:
import openai

results = []
openai.api_key = "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"
j = 0

In [103]:
for num in nums[j:]:
    print(num,data['Label'][num])
    response = openai.ChatCompletion.create(
        model="gpt-3.5-turbo",
        messages=[
                {"role": "system", "content": "You are a Information extraction model that produces rdf triples from the text given, that uses ttl syntax, does not access dbpedia, uses ex prefix, and gives data values to literals."},
                {"role": "user", "content": """output just rdf triples, in ttl format, for the entity '%s' from the text:
    "%s" without using dbo or dbpedia """ %(data['Subjects'][num],data['Abstract'][num])},
            ],
        temperature = 0.6
    )

    result = ''
    for choice in response.choices:
        result += choice.message.content
    results.append(result)
    j+=1
#print(results)

173 Myanmar
20 Germany
145 Russia
146 Philippines
136 Emma Watson
125 Leonardo DiCaprio
123 Chris Evans (actor)
33 Serena Williams
165 Joe Biden
81 Daniel Craig


In [ ]:
subs = []
#for num in nums:
      #subs.append( {'Subject': data['Subjects'][num], '1: Temp 0.6': data['GPT-Triples-Raw'][num] } )

#gptcomparison = pd.DataFrame(subs)
#gptcomparison['5: Temp 0.6'] = results
#gptcomparison.to_csv('GPT-Comparison-0.csv')
#gptcomparison

In [376]:
gptcomparison = pd.read_csv('GPT-Comparison-10.csv')
#gptcomparison = pd.read_csv('GPT-Comp-0-pro.csv')
gptcomparison = gptcomparison.drop('Unnamed: 0', axis=1)
#gptcomparison = gptcomparison.rename(columns={"2: Temp 0.4": "2: Temp 0", "3: Temp 0.4": "3: Temp 0", "4: Temp 0.4": '4: Temp 0'})
gptcomparison

Subject                                          1: Temp 1   
0              Myanmar  @prefix ex: <http://example.com/> .\n@prefix r...  \
1              Germany  @prefix ex: <http://example.com/> .\n\nex:Germ...   
2               Russia  @prefix ex: <http://example.com/> .\n@prefix r...   
3          Philippines  @prefix ex: <http://example.com/> .\n@prefix r...   
4          Emma_Watson  @prefix ex: <http://example.com/> .\n@prefix r...   
5    Leonardo_DiCaprio  @prefix ex: <http://example.com/> .\n@prefix r...   
6  Chris_Evans_(actor)  ```\n@prefix ex: <http://example.com/> .\n\nex...   
7      Serena_Williams  @prefix ex: <http://example.com/> .\n@prefix r...   
8            Joe_Biden  @prefix ex: <http://example.com/> .\n@prefix x...   
9         Daniel_Craig  @prefix ex: <http://example.com/> .\n@prefix r...   

                                           2: Temp 1   
0  @prefix ex: <http://example.com/> .\n\nex:Myan...  \
1  @prefix ex: <http://example.com/> .\n@prefix r...   
2  The text given mentions several aspects relate...   
3  @prefix ex: <http://example.com/> .\n\nex:Phil...   
4  @prefix ex: <http://example.com/> .\n\nex:Emma...   
5  @prefix ex: <http://example.com/> .\n@prefix x...   
6  @prefix ex: <http://example.com/> .\n@prefix r...   
7  @prefix ex: <http://example.com/> .\n@prefix r...   
8  @prefix ex: <http://example.com/> .\n@prefix x...   
9  @prefix ex: <http://example.org/> .\n@prefix x...   

                                           3: Temp 1   
0  @prefix ex: <http://example.com/> .\n\nex:Myan...  \
1  @prefix ex: <http://example.com/> .\n@prefix r...   
2  prefix ex: <http://example.com/>\n\nex:Russia ...   
3  @prefix ex: <http://example.com/> .\n@prefix r...   
4  @prefix ex: <http://example.com/> .\n\nex:Emma...   
5  @prefix ex: <http://example.com/> .\n@prefix r...   
6  ```\n@prefix ex: <http://example.com/> .\n\nex...   
7  @prefix ex: <http://example.com/> .\n@prefix r...   
8  @prefix ex: <http://example.com/> .\n@prefix r...   
9  Prefixes:\n```\n@prefix ex: <http://example.co...   

                                           4: Temp 1   
0  ex:Myanmar rdf:type owl:NamedIndividual ,\n   ...  \
1  @prefix ex: <http://example.com/> .\n\nex:Germ...   
2  prefix ex: <http://example.com/> .\n\nex:Russi...   
3  @prefix ex: <http://example.com/> .\n\nex:Phil...   
4  @prefix ex: <http://example.com/> .\n@prefix r...   
5  @prefix ex: <http://example.com/> .\n@prefix x...   
6  ```\n@prefix ex: <http://example.com/> .\n\nex...   
7  @prefix ex: <http://example.com/> .\n@prefix r...   
8  @prefix ex: <http://example.com/> .\n@prefix x...   
9  @prefix ex: <http://example.com/> .\n@prefix r...   

                                           5: Temp 1  
0  @prefix ex: <http://example.com/> .\n@prefix r...  
1  @prefix ex: <http://example.com/> .\n\nex:Germ...  
2  @prefix ex: <http://example.com/> .\n@prefix r...  
3  @prefix ex: <http://example.com/> .\n\nex:Phil...  
4  @prefix ex: <http://example.com/> .\n@prefix r...  
5  @prefix ex: <http://example.com/> .\n@prefix r...  
6  @prefix ex: <http://example.com/> .\n@prefix r...  
7  @prefix ex: <http://example.com/> .\n\nex:Sere...  
8  @prefix ex: <http://example.com/> .\n@prefix r...  
9  @prefix ex: <http://example.com/> .\n@prefix r...

In [377]:
ent = False
k = 0
dd = []
aa = []
entnamet = 0
allaa = []
for i in range(1,6):
    print(i)
    for s,t in zip(gptcomparison['Subject'], gptcomparison['%s: Temp 1' %(i)]):
    #for s,t in zip(gptcomparison['Subject'], gptcomparison['%s:1' %(i)]):
        print(k,s)
        for i in t.split('\n'):
            if len(i) == 0 or i == '```' or i[0] == '@' or i == 'Prefixes:' or i == 'RDF triples:' or i == '```turtle':
                continue
            if entnamet == 0:
                entname = i.split(' ')[0]
                entnamet += 1
            if (s in i.split(' ')[0] or entname in i.split(' ')[0]) and len(i.split(' ')) > 1:
                bb = []
                #print(i)
                for j in i.split(' '):
                    if len(j) != 0 and j != ';' and j != '.' and j != ',':
                        bb.append(j)
                if i[-1] != ',':   
                    print(bb)
                    dd.append(bb)
                    pred = bb[1]
                    bb = []
                #print(bb)
                #dd.append(bb)
                #bb = []
                ent = True
                cont = i[-1]
                #print(pred)
            elif ent == True and cont == ';' or cont == ',':
                #print('yes')
                if cont == ',':
                    for j in i.split(' '):
                        if len(j) != 0 and j != ';' and j != '.' and j != ',':
                            bb.append(j)
                            if i[-1] == ';':
                                print(bb)
                                dd.append(bb)
                                bb = []
                else:
                    #bb = []
                    #print(i)
                    for j in i.split(' '):
                        if len(j) != 0 and j != ';' and j != '.' and j != ',':
                            bb.append(j)
                    if len(bb) == 1:
                        print('Ye: 1', bb)
                    if i[-1] != ',':   
                        print(bb)
                        dd.append(bb)
                        bb = []
                cont = i[-1]
            else:
                ent = False
                continue
        k +=1
        entnamet = 0
        aa.append(dd)
        dd = []
        print()
    allaa.append(aa)
    aa = []

1
0 Myanmar
['ex:Myanmar', 'rdf:type', 'ex:Country']
['ex:officialName', '"Republic', 'of', 'the', 'Union', 'of', 'Myanmar"^^xsd:string']
['ex:formerName', '"Burma"^^xsd:string']
['ex:locatedIn', 'ex:SoutheastAsia']
['ex:hasCapital', 'ex:Naypyidaw']
['ex:hasLargestCity', 'ex:Yangon']
['ex:hasBorderWith', 'ex:Bangladesh,', 'ex:India,', 'ex:China,', 'ex:Laos,', 'ex:Thailand']
['ex:hasCoastlineWith', 'ex:AndamanSea,', 'ex:BayOfBengal']
['ex:hasPopulation', '"54000000"^^xsd:integer']
['ex:hasArea', '"676578"^^xsd:integer']
['ex:hasLanguage', 'ex:Burmese']
['ex:hasReligion', 'ex:TheravadaBuddhism']
['ex:hasHistory', '[']

1 Germany
['ex:Germany', 'ex:name', '"Germany"']
['ex:Germany', 'ex:officialName', '"Federal', 'Republic', 'of', 'Germany"']
['ex:Germany', 'ex:population', '"84000000"']
['ex:Germany', 'ex:area', '"357022"']
['ex:Germany', 'ex:capital', '"Berlin"']
['ex:Germany', 'ex:largestCity', '"Ruhr"']
['ex:Germany', 'ex:location', '"Central', 'Europe"']
['ex:Germany', 'ex:language',

In [378]:
first = True 
k = 0
count = 1
triples = []
alltrips = []
comparison = []
for aa in allaa:
    print(count)
    for trips in aa:

        #print(k)
        for trip in trips:

            if first is True:
                sub = trip[0]
                first = False

            if trip[0] == sub:
                #print(trip)
                pred = trip[1]

                if len(trip) >= 3:

                    if trip[2][0] == '"':

                        listing = " ".join(trip[2:]).split(', ')

                        if len(listing) > 1 and listing[-1].split(' ')[0] == 'and':

                            for i in " ".join(trip[2:]).split(','):

                                #print('working')
                                obj = i.replace('"', '')
                                obj = obj.replace('and', '')
                                triples.append([sub, trip[1], '"'+obj.lstrip()+'"'])
                                print('1',sub, trip[1], '"'+obj.lstrip()+'"')

                        else:
                        #print('try2:', trip)
                            triples.append([sub, trip[1], " ".join(trip[2:])])
                            print('2',sub, trip[1], " ".join(trip[2:]))
                    else:

                        for i in trip[2:]:
                            triples.append([sub, trip[1], i])
                            print('3',sub, trip[1], i)

            elif trip[0] != sub:

                if len(trip) > 1:

                    pred = trip[0]
                    if trip[1][0] == '"':

                        listing = " ".join(trip[1:]).split(', ')

                        if len(listing) > 1 and listing[-1].split(' ')[0] == 'and':

                            for i in " ".join(trip[1:]).split(','):

                                obj = i.replace('"', '')
                                obj = obj.replace('and', '')
                                triples.append([sub, trip[0], '"'+obj.lstrip()+'"'])
                                print('4',sub, trip[0], '"'+obj.lstrip()+'"')

                        elif len(listing) > 1 and 'xsd:' in listing[-1]:
                            for i in listing:
                                triples.append([sub, trip[0], i])
                                print('5',[sub, trip[0], i])

                        else:
                            triples.append([sub, trip[0], " ".join(trip[1:])])
                            print('6',[sub, trip[0], " ".join(trip[1:])])

                    else:
                        for i in trip[1:]:
                            triples.append([sub, trip[0], i])
                            print('7',[sub, trip[0], i])
                else:
                    #print('ye:')
                    triples.append([sub, pred, trip[0]])
                    print('8',[sub, pred, trip[0]])

        first = True
        print()
        k+=1
        alltrips.append(triples)
        triples = []
    comparison.append(alltrips)
    alltrips = []
    count += 1

1
3 ex:Myanmar rdf:type ex:Country
6 ['ex:Myanmar', 'ex:officialName', '"Republic of the Union of Myanmar"^^xsd:string']
6 ['ex:Myanmar', 'ex:formerName', '"Burma"^^xsd:string']
7 ['ex:Myanmar', 'ex:locatedIn', 'ex:SoutheastAsia']
7 ['ex:Myanmar', 'ex:hasCapital', 'ex:Naypyidaw']
7 ['ex:Myanmar', 'ex:hasLargestCity', 'ex:Yangon']
7 ['ex:Myanmar', 'ex:hasBorderWith', 'ex:Bangladesh,']
7 ['ex:Myanmar', 'ex:hasBorderWith', 'ex:India,']
7 ['ex:Myanmar', 'ex:hasBorderWith', 'ex:China,']
7 ['ex:Myanmar', 'ex:hasBorderWith', 'ex:Laos,']
7 ['ex:Myanmar', 'ex:hasBorderWith', 'ex:Thailand']
7 ['ex:Myanmar', 'ex:hasCoastlineWith', 'ex:AndamanSea,']
7 ['ex:Myanmar', 'ex:hasCoastlineWith', 'ex:BayOfBengal']
6 ['ex:Myanmar', 'ex:hasPopulation', '"54000000"^^xsd:integer']
6 ['ex:Myanmar', 'ex:hasArea', '"676578"^^xsd:integer']
7 ['ex:Myanmar', 'ex:hasLanguage', 'ex:Burmese']
7 ['ex:Myanmar', 'ex:hasReligion', 'ex:TheravadaBuddhism']
7 ['ex:Myanmar', 'ex:hasHistory', '[']

2 ex:Germany ex:name "German

In [379]:
triples = []
subject = []
version = []
for i in comparison:
    for j in i:
        for k in j:
            if '^^xsd:string' in k[-1]:
                print(k[-1])
                print(k[-1].replace("^^xsd:string", ""))
                print()
                triples.append([k[0], k[1], k[-1].replace("^^xsd:string", "")])
            else:
                triples.append(k)
        subject.append(triples)
        triples = []
    version.append(subject)
    subject = []

comparison = version
                
                

"Republic of the Union of Myanmar"^^xsd:string
"Republic of the Union of Myanmar"

"Burma"^^xsd:string
"Burma"

"Los Angeles"^^xsd:string
"Los Angeles"

"American"^^xsd:string
"American"

"Academy Award"^^xsd:string
"Academy Award"

"British Academy Film Award"^^xsd:string
"British Academy Film Award"

"Golden Globe Award"^^xsd:string
"Golden Globe Award"

"eight times in annual rankings of the world's highest-paid actors"^^xsd:string
"eight times in annual rankings of the world's highest-paid actors"

"Leonardo DiCaprio Foundation"^^xsd:string
"Leonardo DiCaprio Foundation"

"3.8 trillion USD"^^xsd:string
"3.8 trillion USD"

"Serena Jameka Williams"^^xsd:string
"Serena Jameka Williams"

"American"^^xsd:string
"American"

"Singles"^^xsd:string
"Singles"

"Women's Tennis Association (WTA)"^^xsd:string
"Women's Tennis Association (WTA)"

"Year-End No. 1"^^xsd:string
"Year-End No. 1"

"Open Era"^^xsd:string
"Open Era"

"US Open"^^xsd:string
"US Open"

"Major Singles"^^xsd:string
"Major Si

In [380]:
count = 1
for sub in comparison:
    if count == 1:
        comparisondata = pd.DataFrame({'Subject': gptcomparison['Subject'], '%s:1' %(count): sub})
    else:
        comparisondata['%s:1' %(count)] = sub
    count += 1
comparisondata
    

Subject                                                1:1   
0              Myanmar  [[ex:Myanmar, rdf:type, ex:Country], [ex:Myanm...  \
1              Germany  [[ex:Germany, ex:name, "Germany"], [ex:Germany...   
2               Russia  [[ex:Russia, rdf:type, ex:Country], [ex:Russia...   
3          Philippines  [[ex:Philippines, rdf:type, ex:Country], [ex:P...   
4          Emma_Watson  [[ex:Emma_Watson, rdf:type, ex:Person], [ex:Em...   
5    Leonardo_DiCaprio  [[ex:Leonardo_DiCaprio, rdf:type, foaf:Person]...   
6  Chris_Evans_(actor)  [[ex:Chris_Evans_(actor), a, ex:Actor], [ex:Ch...   
7      Serena_Williams  [[ex:Serena_Williams, rdf:type, ex:TennisPlaye...   
8            Joe_Biden  [[ex:Joe_Biden, a, ex:Person], [ex:Joe_Biden, ...   
9         Daniel_Craig  [[ex:daniel_craig, rdf:type, ex:Actor], [ex:da...   

                                                 2:1   
0  [[ex:Myanmar, ex:officialName, "Republic of th...  \
1  [[ex:Germany, rdf:type, ex:Country], [ex:Germa...   
2  [[The, text, given], [The, text, mentions], [T...   
3  [[ex:Philippines, ex:hasOfficialName, "Republi...   
4  [[ex:Emma_Watson, a, ex:Actress], [ex:Emma_Wat...   
5  [[ex:Leonardo_DiCaprio, rdf:type, ex:Actor], [...   
6  [[ex:Chris_Evans, rdf:type, ex:Actor], [ex:Chr...   
7  [[ex:Serena_Williams, rdf:type, ex:TennisPlaye...   
8  [[ex:Joe_Biden, rdf:type, ex:Person], [ex:Joe_...   
9  [[ex:Daniel_Craig, a, ex:Actor], [ex:Daniel_Cr...   

                                                 3:1   
0  [[ex:Myanmar, a, ex:Country], [ex:Myanmar, ex:...  \
1  [[ex:Germany, ex:hasOfficialName, "Federal Rep...   
2  [[prefix, ex:, <http://example.com/>], [prefix...   
3  [[ex:Philippines, a, ex:Country;], [ex:Philipp...   
4  [[ex:Emma_Watson, ex:name, "Emma Charlotte Due...   
5  [[ex:Leonardo_DiCaprio, rdf:type, ex:Person], ...   
6  [[ex:Chris_Evans_(actor), a, ex:Actor], [ex:Ch...   
7  [[ex:Serena_Williams, rdf:type, ex:TennisPlaye...   
8  [[ex:Joe_Biden, rdf:type, ex:Person], [ex:Joe_...   
9  [[ex:Daniel_Craig, rdf:type, ex:Actor], [ex:Da...   

                                                 4:1   
0  [[ex:Myanmar, rdf:type, owl:NamedIndividual], ...  \
1  [[ex:Germany, a, ex:Country], [ex:Germany, ex:...   
2  [[prefix, ex:, <http://example.com/>], [prefix...   
3  [[ex:Philippines, ex:hasName, "Philippines"], ...   
4  [[ex:Emma_Watson, rdf:type, ex:Actor], [ex:Emm...   
5  [[#, Leonardo, DiCaprio], [#, ex:Leonardo_DiCa...   
6  [[ex:Chris_Evans_(actor), a, ex:Actor], [ex:Ch...   
7  [[ex:Serena_Williams, rdf:type, ex:TennisPlaye...   
8  [[ex:Joe_Biden, rdf:type, ex:Person], [ex:Joe_...   
9  [[ex:Daniel_Craig, rdf:type, ex:Actor], [ex:Da...   

                                                 5:1  
0  [[ex:Myanmar, rdf:type, ex:Country], [ex:Myanm...  
1  [[ex:Germany, a, ex:Country], [ex:Germany, ex:...  
2  [[ex:Russia, rdf:type, ex:Country], [ex:Russia...  
3  [[ex:Philippines, a, ex:Country], [ex:Philippi...  
4  [[ex:Emma_Watson, rdf:type, ex:Person], [ex:Em...  
5  [[ex:Leonardo_DiCaprio, rdf:type, ex:Actor], [...  
6  [[ex:Chris_Evans, rdf:type, ex:Actor], [ex:Chr...  
7  [[ex:Serena_Williams, a, ex:TennisPlayer], [ex...  
8  [[ex:Joe_Biden, rdf:type, ex:Person], [ex:Joe_...  
9  [[ex:Daniel_Craig, rdf:type, ex:Actor], [ex:Da...

In [381]:
import ast
same = 0
notsame = 0
count = 0
for subnum in range(0,5):
    #print(subnum)
    #print(len(comparisondata['1:1'][subnum]))
    count += len(comparisondata['1:1'][subnum])
    version = [comparisondata['1:1'][subnum],comparisondata['2:1'][subnum],comparisondata['3:1'][subnum],comparisondata['4:1'][subnum],comparisondata['5:1'][subnum]]
    version.sort(key=len)
    for i in version[0]:
        #for j in comparisondata['2:1.6'][subnum]:
        if i in version[1] and i in version[2] and i in version[3] and i in version[4]:
            print(i)
            same += 1
            continue
listsame = []   
check = 0
listcheck = []
for subnum in range(0,5):
    for num in range(1,6):
        for i in comparisondata['%s:1' %(num)][subnum]:
            if i not in listsame:
                notsame += 1
                listsame.append(i)
            else:
                if i not in listcheck:
                    check += 1
                    listcheck.append(i)
same, notsame, count, check      

(0, 957, 191, 75)

In [400]:
round((147/1103)*100,2)

13.33

In [328]:
#comparisondata.to_csv('GPT-Comp-10-pro.csv')


temp 0
- 63, 437, 222
- 14.4% JI
- 50.8% Mult
- variety match over the 5 outputs 
- data values and links 

temp 0.2
- 20, 848, 140 
- 2.4% JI
- 16.5% mult
- mainly countries 
- 2 for actor 

temp 0.4
- 4, 867, 156
- 0.5% JI
- 18% mult
- only countries 

temp 0.6
- 1, 998, 116
- 0.1 JI
- 11.6% mult
- only one country

temp 0.8
- 1, 1103, 147
- 0.09% JI
- 13.33% mult
- country again 

temp 1
- 0, 957, 75
- 0% JI
- 7.7% mult
- no matches 


By looking at the jaccards index of all the different temperatures it can be seen that all the models have a low score. Even when the temperature is turned down to 0, making the model the most confident in its output, restricting the creativity of the output, only has a jaccard score of 14.4%. This means that only 14.4% of all the triples in the five outputs of the zero temperautre occured in all the outputs, this is still better than all the rest and inparticular temp 1, where 0% of the triples were repeated in all. This comments on the spontainty and unnrealiability of chatGPTs results no matter the temperature. 

to determine if the triples repeated at all, the amount of triple more than once was also recorded and this had a huge difference in the jaccard index score. 53% of the triples in the temperature zero model occured more than once, this demonstrates the ability of the triples to repeat, just not on a reliable level. There were a few anomolies in the results where in some of the high temperature models having more triples with multiple occurrances than models with a temperature below, this could be due the the higher temperature models in general producing larger amounts of triples.